AutoCollection/ClientRequestParser.ts

```diff
@@ -48,15 +53,26 @@ class ClientRequestParser extends RequestParser {
         let dependencyName = this.method.toUpperCase() + " " + url.format(urlObject);
 
         let remoteDependency = new ContractsModule.Contracts.RemoteDependencyData();
-        remoteDependency.target = urlObject.hostname;
+
+        if (this.targetIKeyHash) {
+            // Currently the service supports a dependency type name of ApplicationInsights,
+            // but not a dependency kind. So the dependency kind here is still just HTTP.
+            remoteDependency.dependencyKind = ContractsModule.Contracts.DependencyKind.Http;
+            remoteDependency.dependencyTypeName = ContractsModule.Contracts.DependencyKind
+                [ContractsModule.Contracts.DependencyKind.ApplicationInsights];
```

Comment:
Actually no. Per the latest schema DependencyKind is deprecated. We shouldn't be sending enumerated values anymore. We should be using the type field which takes a string value.